### ISSUE TITLE
Inconsistent results for KeyExtractor and EqualLoudness on consecutive runs without reset()

### ISSUE THREAD

**Issue by krasch** [MTG/essentia#1341]
(This issue is modeled after https://github.com/MTG/essentia/issues/1324)

KeyExtractor in standard mode generates incorrect outputs on consecutive compute() runs. Running reset() after each run fixes the problem:

```
from essentia.standard import KeyExtractor

extractor = KeyExtractor()
result = extractor(signal1)
print(f"signal1: {result}")

result = extractor(signal2)
print(f"signal2 without reset(): {result}")

extractor.reset()
result = extractor(signal2)
print(f"signal2 with reset(): {result}")

```

Output:

```
signal1: ('E', 'minor', 0.8703092932701111)
signal2 without reset(): ('E', 'minor', 0.8417810201644897)
signal2 with reset(): ('F', 'major', 0.8114738464355469)
```

Something similar is happening with EqualLoudness:

```
from essentia.standard import EqualLoudness

loudness = EqualLoudness()
result = loudness(signal1)
print(f"signal1: {result[0:3]}")

result = loudness(signal2)
print(f"signal2 without reset(): {result[0:3]}")

loudness.reset()
result = loudness(signal2)
print(f"signal2 with reset(): {result[0:3]}")

```

Output:

```
signal1: [-6.3488181e-07 -2.2986151e-06 -3.5950634e-06]
signal2 without reset(): [-4.3279724e-06 -3.4251239e-06 -1.0023520e-06]
signal2 without reset(): [-1.3271563e-07 -2.8904444e-07  2.6350062e-07]
```

On the other hand, for EqualLoudness, the unit tests seem to suggest that it is the callers responsibility to call `.reset()`: https://github.com/MTG/essentia/blob/master/test/src/unittests/filters/test_equalloudness.py#L44

We are wondering, is this behaviour a bug? Or is it expected, and we should always make sure to call `.reset()` after using the extractor/filter, when a `.reset()` method is available? 

Thank you so much!

**Comment by palonso** [comment]
Hi @krasch,
right now the user is expected to `.reset()` these algorithms between successive calls. However, I don't see how automatically resetting them at the end of each stream would hurt, so I'm in favor of implementing this feature.

@dbogdanov, do you see any reason not to do it?

**Comment by dbogdanov** [comment]
In the case of EqualLoudness, this is because [StreamingAlgorithmWrapper](https://github.com/MTG/essentia/blob/master/src/algorithms/filters/equalloudness.h#L82) needs to maintain [filter state](https://github.com/MTG/essentia/blob/master/src/algorithms/filters/iir.cpp#L34) between consequent computes. It is consistent with the rest of the filter algorithms. We should clarify this behavior better.

**Comment by dbogdanov** [comment]
The behavior for KeyExtractor is fixed in #1341.

**Comment by krasch** [comment]
@dbogdanov Thank you so much for that bugfix, KeyExtractor works as expected for us now

And we'll make sure to call .reset() on any filters inbetween tracks.

I guess this issue can be closed now?
